src/neml2/models/solid_mechanics/IsotropicElasticityTensor.cxx

```diff
@@ -0,0 +1,127 @@
+// Copyright 2024, UChicago Argonne, LLC
+// All Rights Reserved
+// Software Name: NEML2 -- the New Engineering material Model Library, version 2
+// By: Argonne National Laboratory
+// OPEN SOURCE LICENSE (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include "neml2/models/solid_mechanics/IsotropicElasticityTensor.h"
+
+namespace neml2
+{
+register_NEML2_object(IsotropicElasticityTensor);
+
+OptionSet
+IsotropicElasticityTensor::expected_options()
+{
+  OptionSet options = ElasticityTensor::expected_options();
+  options.doc() += "  This class defines an isotropic elasticity tensor using two parameters."
+                   "  Various options are available for which two parameters to provide.";
+
+  options.set_parameter<CrossRef<Scalar>>("p1");
+  options.set("p1").doc() = "First parameter";
+
+  EnumSelection type_selection({"youngs_modulus", "poissons_ratio", "INVALID"},
+                               {static_cast<int>(IsotropicElasticityTensor::ParamType::YOUNGS),
+                                static_cast<int>(IsotropicElasticityTensor::ParamType::POISSONS),
+                                static_cast<int>(IsotropicElasticityTensor::ParamType::INVALID)},
+                               "INVALID");
+  options.set<EnumSelection>("p1_type") = type_selection;
+  options.set("p1_type").doc() =
+      "First parameter type. Options are: " + type_selection.candidates_str();
+
+  options.set_parameter<CrossRef<Scalar>>("p2");
+  options.set("p2").doc() = "Second parameter";
+
+  options.set<EnumSelection>("p2_type") = type_selection;
+  options.set("p2_type").doc() =
+      "Second parameter type. Options are: " + type_selection.candidates_str();
```

Comment:
Similarly this can be changed to a list of parameter types + a list of parameter values.

Another advantage of this approach is that we could factor out a common base class for cubic and isotropic elasticity tensors.